packages/html/src/layers/htmlImageLayer.ts

```diff
@@ -28,8 +29,9 @@ export class HtmlImageLayer{
   update(userCloudinaryImage: CloudinaryImage, plugins: any, analyticsOptions?: AnalyticsOptions){
     const pluginCloudinaryImage  = cloneDeep(userCloudinaryImage);
     render(this.imgElement, pluginCloudinaryImage, plugins, this.htmlPluginState)
-        .then(()=>{
-          this.imgElement.setAttribute('src', pluginCloudinaryImage.toURL(getAnalyticsOptions(analyticsOptions)));
+        .then((pluginResponse)=>{
+            const featuredAnalyticsOptions = getAnalyticsOptions(analyticsOptions, pluginResponse);
```

Comment:
If you rename the `FeaturedAnalyticsOptions` to `PluginResponse` then this could be simplified to

```suggestion
            const analyticsOptions = getAnalyticsOptions(analyticsOptions, pluginResponse);
```